File: src/rotation_listener.py
import time

# DATA_PER_INTERVAL = 5;
INTERVAL = 0.1;

class RotationListener(object):
  UP, DOWN, LEFT, RIGHT = 0, 1, 2, 3

  def __init__(self):
    self.prev_avg = [0.0, 0.0]
    self.curr_data = []
    self.motion_threshold = 2.0
    self.zero_threshold = 0.5
    self.can_move = True
    with open("data/curr_direction.txt", "w+") as f:
      f.write("3"); # start off going to right

  def refresh_data(self):
    with open("data/accel_data.txt", "r+") as f:
      self.curr_data = f.readlines()

  def listen(self):
    while True:
      # time.sleep(INTERVAL)
      raw_input("...")
      self.refresh_data()
      curr_avg = [0.0, 0.0]
      for datum_str in self.curr_data:
        datum = datum_str.split()
        if datum != "[]":
          curr_avg = [curr_avg[i] + (float)(datum[i]) for i in range(2)]
      l = len(self.curr_data)
      curr_avg = [a / l for a in curr_avg]
      print(curr_avg)
      print(self.prev_avg)
      # figure out which direction the phone moved
      if ((abs(curr_avg[0]) < zero_threshold) and (abs(curr_avg[1]) < zero_threshold)):
        self.can_move = True
      else:
        for i in range(len(curr_avg)):
          if abs(curr_avg[i]) > self.motion_threshold:
            if curr_avg[i] > self.motion_threshold:
              if i == 0:
                # it moved in the right direction
                print("right")
                if can_move:
                  self.register_direction(RotationListener.RIGHT)
                self.can_move = False
              else:
                # it moved in the up direction
                print("up")
                if can_move:
                  self.register_direction(RotationListener.DOWN)
                self.can_move = False
            else:
              if i == 0:
                # it moved in the left direction
                print("left")
                if can_move:
                  self.register_direction(RotationListener.LEFT)
                self.can_move = False
              else:
                # it moved in the down direction
                print("down")
                if can_move:
                  self.register_direction(RotationListener.DOWN)
                self.can_move = False
      self.prev_avg = curr_avg

  def register_direction(self, direction):
    with open("../data/curr_direction.txt", "w+") as f:
      f.write(str(direction))

if __name__ == "__main__":
  RotationListener().listen()

File: src/RotationListener.java
package de.vogella.android.sensor;

import java.util.ArrayList;
import java.lang.Math;

public void RotationListener(Array positions) {
  int UP, DOWN, LEFT, RIGHT = 0, 1, 2, 3;
  ArrayList<Double> currData;
  float motionThreshold = 2.0;
  float zeroThreshold = 0.5;
  boolean canMove = true;
  double[] currAvg;
  currData = new ArrayList<Double>();

  //write 3 to curr_direction
  BufferedWriter out = null;
  try {
    FileWriter fstream = new FileWriter("data/curr_direction.txt", false);
    out = new BufferedWriter(fstream);
    out.write("3");
   }
   catch (IOException e) {
     System.err.println("Error: " + e.getMessage());
   }
   finally {
     if(out != null) {
       out.close();
     }
   }

  while true:
    refreshData();
    currAvg = {0.0, 0.0};
    Double total = 0.0;
    for (int i = 0; i < 2; i++) {
      for (int j = 0; j < 10; j++) {
        total += currAvg[j][i]; 
      }
    currAvg[i] = total/10;
    }
    for (Double d : currAvg) {
      System.out.println(d);
    }
    if ((Math.abs(currAvg[0]) < zeroThreshold) & (Math.abs(currAvg[1]) < zeroThreshold) {
      canMove = true;
    } else {
      for (i = 0; i < currAvg.length(); i++) {
        if (Math.abs(currAvg[i]) > motionThreshold) {
          if (currAvg[i] > motionThreshold) {
            if (i == 0) {
              System.out.println("right");
              if (canMove) {
                registerDirection(RIGHT);
              }
              canMove = false;
            } else {
              System.out.println("up");
              if (canMove) {
                registerDirection(UP);
              }
              canMove = false;
            }
          } else {
            if (i == 0) {
              System.out.println("left");
              if (canMove) {
                registerDirection(LEFT);
              }
              canMove = false;
            } else {
              System.out.println("down");
              if (canMove) {
                registerDirection(DOWN);
              }
              canMove = false;
            }
          }
        } 
      }
    }

  private void registerDirection(int direction) {
  	out = null;
  try {
    FileWriter fstream = new FileWriter("data/curr_direction.txt", false);
    out = new BufferedWriter(fstream);
    out.write(direction);
   }
   catch (IOException e) {
     System.err.println("Error: " + e.getMessage());
   }
   finally {
     if(out != null) {
       out.close();
     }
   }
  }

  private void refreshData() {
    File file = new File("data/pos_data.txt");
    BufferedReader reader = null;

    try {
      reader = new BufferedReader(new FileReader(file));
      String text = null;

      while ((text = reader.readLine()) != null) {
        currData.add(text);
      }
    } catch (FileNotFoundException e) {
      e.printStackTrace();
    } catch (IOException e) {
      e.printStackTrace();
    } finally {
      try {
        if (reader != null) {
          reader.close();
        }
      } catch (IOException e) {
      }
    }
  }
}






